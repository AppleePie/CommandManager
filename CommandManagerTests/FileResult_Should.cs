using System;
using System.IO;
using System.Linq;
using CommandManager;
using CommandManager.Contracts;
using CommandManager.Infrastructure;
using CommandManager.Results;
using FakeItEasy;
using FluentAssertions;
using NUnit.Framework;

namespace CommandManagerTests
{
    
    [TestFixture]
    public class FileResult_Should
    {
        private IExecutor executor;
        private readonly FileResult resulter = new FileResult();

        [SetUp]
        public void SetUp()
        {
            executor = A.Fake<IExecutor>();
            A.CallTo(() => executor.Execute("test")).Returns(Result.Ok("Done!"));
        }

        [Test]
        public void DumpResult_EmptyResults_EmptyOutput()
        {
            resulter.DumpResult(Enumerable.Empty<Result<string>>());
            File.ReadAllText(FileResult.FileResultPath).Should().Be(Environment.NewLine);
            File.Delete(FileResult.FileResultPath);
        }

        [Test]
        public void DumpResult_RunCommand_CorrectResultInFile()
        {
            resulter.DumpResult(Enumerable.Repeat(executor.Execute("test"), 1));
            File.ReadAllText(FileResult.FileResultPath).Should().Be("Done!" + Environment.NewLine + Environment.NewLine);
            File.Delete(FileResult.FileResultPath);
        }
    }
}