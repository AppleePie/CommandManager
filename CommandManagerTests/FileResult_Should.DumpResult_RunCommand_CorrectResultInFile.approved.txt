MD5 hash for D:\Tasks\CommandManager\CommandManagerTests\bin\Debug\netcoreapp3.1\TestDir\NewFile.txt: c7d96398128a3d927be521fe379d1253
Total MD5 hash for directory D:\Tasks\CommandManager\CommandManagerTests\bin\Debug\netcoreapp3.1\TestDir: c7d96398128a3d927be521fe379d1253
